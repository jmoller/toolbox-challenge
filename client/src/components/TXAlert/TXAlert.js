import React from "react";
import { Alert } from "react-bootstrap";
import Proptypes from 'prop-types';

const TXAlert = ({ message, errors }) => {
  const alertStyle = errors === undefined ? "success" : "danger";
  const alertText = errors === undefined ? "Response From Server:" : "Error: ";
  const alertDescription = errors === undefined ? message : errors;
  return (
    <Alert bsStyle={alertStyle} style={{ width: "25rem", marginTop: "2rem" }}>
      <strong>{alertText}</strong>
      <p>{alertDescription}</p>
    </Alert>
  );
};

TXAlert.propTypes = {
  message: Proptypes.string,
  errors: Proptypes.string
}

export default TXAlert;
