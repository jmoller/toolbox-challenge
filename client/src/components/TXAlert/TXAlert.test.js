import React from "react";
import { shallow, mount } from "enzyme";
import TXAlert from "./TXAlert";

describe("TXAlert", () => {
  const txalert = shallow(<TXAlert />);

  it("renders TXAlert correctly", () => {
    expect(txalert).toMatchSnapshot();
  });

  describe("the user submits a message", () => {
    let txalertWithProps = "";
    beforeEach(() => {
      const props = { message: "This is a mock message" };
      txalertWithProps = shallow(<TXAlert {...props} />);
    });

    it("displays the alert with the `success` class attaches to it", () => {
      expect(txalertWithProps.props().bsStyle).toEqual("success");
    });

    it("displays the alert with the title equal to `Success!`", () => {
      const expectedTitle = "Response From Server:";
      expect(txalertWithProps.find("strong").text()).toEqual(expectedTitle);
    });

    it("displays the alert description with the message properly set into it", () => {
      const expectedDescription = "This is a mock message";
      expect(txalertWithProps.find("p").text()).toEqual(expectedDescription);
    });

    
  });

  describe("the user submits a an empty message", () => {
    let txalertWithProps = "";
    beforeEach(() => {
      const props = { errors: "This is a mock error" };
      txalertWithProps = shallow(<TXAlert {...props} />);
    });

    it("displays the alert with the `danger` class attaches to it", () => {
      expect(txalertWithProps.props().bsStyle).toEqual("danger");
    });

    it("displays the alert with the title equal to `Error`", () => {
      const expectedTitle = "Error: ";
      expect(txalertWithProps.find("strong").text()).toEqual(expectedTitle);
    });

    it("displays the alert description with the message properly set into it", () => {
      const expectedDescription = "This is a mock error";
      expect(txalertWithProps.find("p").text()).toEqual(expectedDescription);
    });
  });
});
