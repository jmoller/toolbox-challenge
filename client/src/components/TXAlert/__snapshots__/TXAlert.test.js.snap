// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TXAlert renders TXAlert correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TXAlert />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "alert",
      "bsStyle": "success",
      "children": Array [
        <strong>
          Response From Server:
</strong>,
        <p />,
      ],
      "closeLabel": "Close alert",
      "style": Object {
        "marginTop": "2rem",
        "width": "25rem",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Response From Server:",
        },
        "ref": null,
        "rendered": "Response From Server:",
        "type": "strong",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "alert",
        "bsStyle": "success",
        "children": Array [
          <strong>
            Response From Server:
</strong>,
          <p />,
        ],
        "closeLabel": "Close alert",
        "style": Object {
          "marginTop": "2rem",
          "width": "25rem",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Response From Server:",
          },
          "ref": null,
          "rendered": "Response From Server:",
          "type": "strong",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
